apps/src/templates/NavigationBar.jsx

```diff
@@ -1,42 +1,100 @@
 import React, { PropTypes } from 'react';
+import _ from 'lodash';
 import color from "@cdo/apps/util/color";
+import FontAwesome from './FontAwesome';
 import NavigationBarLink from './NavigationBarLink';
 
 const styles = {
   navBar: {
     height: 50,
     width: '100%',
     backgroundColor: color.purple,
-    marginBottom: 20
+    marginBottom: 20,
+  },
+  linksBox: {
+    overflow: 'auto',
+    whiteSpace: 'nowrap',
+    display: 'inline',
+    float: 'left',
+    width: 920
   },
   linkBox: {
     display: 'inline',
     paddingRight: 20,
     paddingLeft: 20,
   },
+  chevron: {
+    fontSize: 20,
+    lineHeight: '50px',
+    color: color.white,
+    textDecoration: 'none',
+    cursor: 'pointer',
+    height: 50,
+    width: 25,
+    backgroundColor: color.purple,
+    paddingLeft: 10,
+    paddingRight: 10,
+    display: 'inline',
+    float: 'left'
+  }
 };
 
 export default class NavigationBar extends React.Component {
   static propTypes = {
     links: PropTypes.array,
   };
 
+  state = {
+    listPosition: "start"
+  };
+
+  scrollForward = () => {
+    const lastLink = _.last(this.props.links);
+    const element = document.getElementById(lastLink.url);
```

Comment:
since the `this.props.links` prop isn't required, or we may be passed an empty array, i think we want to put a guard around scrolling functionality (here and in `scrollBackward()`). something like:
```
if (lastLink && lastLink.url) {
  const element = document.getElementById(lastLink.url);
  element.scrollIntoView({block: "end"});
  this.setState({listPosition: "end"});
}
```